MapboxNavigation/RouteVoiceController.swift

```diff
@@ -275,4 +277,12 @@ public protocol VoiceControllerDelegate {
      */
     @objc(voiceController:didInterruptSpokenInstruction:withInstruction:)
     optional func voiceController(_ voiceController: RouteVoiceController, didInterrupt interruptedInstruction: SpokenInstruction, with interruptingInstruction: SpokenInstruction)
+    
+    /** Called when a spoken is about to speak. Useful if it is necessary to give a custom instruction instead. Noting, changing the `distanceAlongStep` property on `SpokenInstruction` will have no impact on when the instruction will be said.
+     
+     - parameter voiceController: The voice controller that experienced the interruption.
+     - parameter instruction: The spoken instruction that will be said.
+     **/
+    @objc(voiceController:willSpeakSpokenInstruction:)
+    optional func voiceController(_ voiceController: RouteVoiceController, willSpeak instruction: SpokenInstruction) -> SpokenInstruction?
```

Comment:
The arguments to this method don’t provide enough context for a typical application to intelligently replace the spoken instruction with a different instruction. For that, the implementation would need access to the RouteProgress, the one that comes in the `routeControllerDidPassSpokenInstructionPoint` notification.